example/index.tsx

```diff
@@ -20,20 +21,27 @@ class Example extends React.Component<Props, State> {
         super(props);
         this.state = {
             text: "000000000",
+            isLoading: false
         };
     }
 
     private handleAfterPrint = () => console.log('after print!');
-    private handleBeforePrint = () => console.log('before print!');
+    private handleBeforePrint = () => this.setState({ isLoading: false});
     private renderContent = () => this.componentRef;
     private renderTrigger = () => <button type="button">Print this out!</button>;
-    private onBeforeGetContent = () => this.setState({text: "text changed"});
+    private onBeforeGetContent = () => {
```

Comment:
You'll have to return a Promise here, since `react-to-print` is not aware of when the `setState` call (which can be asynchronous) completes.

```js
return new Promise((resolve) => {
  this.setState({ text: "text changed", isLoading: true }, resolve);
});
```